src/pal/pal_noalloc.h

```diff
@@ -17,17 +17,18 @@ namespace snmalloc
    * address-space manager is initialised with all of the memory that it will
    * ever use.
    *
-   * It takes an error handler delegate as a template argument. This is
+   * It takes an underlying PAL delegate as a template argument. This is
    * expected to forward to the default PAL in most cases.
    */
-  template<typename ErrorHandler>
+  template<SNMALLOC_CONCEPT(ConceptPAL) UnderlyingPAL>
   struct PALNoAlloc
   {
     /**
      * Bitmap of PalFeatures flags indicating the optional features that this
      * PAL supports.
      */
-    static constexpr uint64_t pal_features = NoAllocation;
+    static constexpr uint64_t pal_features = NoAllocation |
```

Comment:
Requiring `pal_features` from the error handler is not ideal.  Can we add an additional optional template parameter for the entropy source, if we want to provide one?  Alternatively, a subclass of this could provide an entropy source if necessary.